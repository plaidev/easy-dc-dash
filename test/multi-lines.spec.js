import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import MultiLines from '@/multi-lines.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('multi-lines', () => {
  Store.registerData([
    { date: new Date('2017-05-18 10:00'), d1: 'a', d2: true, v1: 1, v2: 10 },
    { date: new Date('2017-05-19 10:00'), d1: 'a', d2: false, v1: 2, v2: 20 },
    { date: new Date('2017-05-20 10:00'), d1: 'b', d2: false, v1: 3, v2: 30 },
    { date: new Date('2017-05-21 10:00'), d1: 'b', d2: true, v1: 5, v2: 50 }
  ]);

  const wrapper = mount(localVue.extend(MultiLines), {
    localVue,
    propsData: {
      id: 'test-multi-lines',
      scale: 'time.hour',
      dimension: 'd.date',
      reduce: '[d.v1, d.v2]'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
