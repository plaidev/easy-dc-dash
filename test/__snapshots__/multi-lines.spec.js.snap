// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi-lines to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-multi-lines" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g>
                            <g class="grid-line horizontal" transform="translate(40,20)"></g>
                            <g class="grid-line vertical" transform="translate(40,20)">
                                <line x1="3.333333333333333" y1="150" x2="3.333333333333333" y2="0" opacity="0"></line>
                                <line x1="13.333333333333332" y1="150" x2="13.333333333333332" y2="0" opacity="0"></line>
                                <line x1="23.333333333333332" y1="150" x2="23.333333333333332" y2="0" opacity="0"></line>
                                <line x1="33.333333333333336" y1="150" x2="33.333333333333336" y2="0" opacity="0"></line>
                                <line x1="43.333333333333336" y1="150" x2="43.333333333333336" y2="0" opacity="0"></line>
                                <line x1="53.33333333333333" y1="150" x2="53.33333333333333" y2="0" opacity="0"></line>
                                <line x1="63.333333333333336" y1="150" x2="63.333333333333336" y2="0" opacity="0"></line>
                                <line x1="73.33333333333334" y1="150" x2="73.33333333333334" y2="0" opacity="0"></line>
                                <line x1="83.33333333333333" y1="150" x2="83.33333333333333" y2="0" opacity="0"></line>
                                <line x1="93.33333333333333" y1="150" x2="93.33333333333333" y2="0" opacity="0"></line>
                                <line x1="103.33333333333334" y1="150" x2="103.33333333333334" y2="0" opacity="0"></line>
                                <line x1="113.33333333333333" y1="150" x2="113.33333333333333" y2="0" opacity="0"></line>
                            </g>
                            <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-multi-lines-clip)"></g>
                            <g class="sub _0">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-multi-lines-clip)">
                                    <g class="stack-list">
                                        <g class="stack _0">
                                            <path class="line" stroke="#1f77b4"></path>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                        </g>
                                    </g>
                                    <g class="dc-tooltip-list">
                                        <g class="dc-tooltip _0">
                                            <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                        </g>
                                    </g>
                                </g>
                            </g>
                            <g class="sub _1">
                                <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-multi-lines-clip)">
                                    <g class="stack-list">
                                        <g class="stack _0">
                                            <path class="line" stroke="#ff7f0e"></path>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                            <text class="lineLabel" text-anchor="middle"></text>
                                        </g>
                                    </g>
                                    <g class="dc-tooltip-list">
                                        <g class="dc-tooltip _0">
                                            <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                            <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#ff7f0e"></circle>
                                        </g>
                                    </g>
                                </g>
                            </g>
                            <g class="axis x" transform="translate(40,170)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(3.333333333333333,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(13.333333333333332,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(23.333333333333332,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Fri 19</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(33.333333333333336,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(43.333333333333336,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(53.33333333333333,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(63.333333333333336,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">Sat 20</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(73.33333333333334,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(83.33333333333333,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">12 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(93.33333333333333,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 PM</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(103.33333333333334,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">May 21</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(113.33333333333333,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">06 AM</text>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="axis y" transform="translate(40,20)">
                                <path class="domain"></path>
                            </g>
                        </g>
                        <defs>
                            <clippath id="test-multi-lines-clip">
                                <rect width="120" height="150" transform="translate(-0, -0)"></rect>
                            </clippath>
                        </defs>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
